cli/help.go

```diff
@@ -0,0 +1,244 @@
+package cli
+
+import (
+	"bufio"
+	"bytes"
+	_ "embed"
+	"flag"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"text/template"
+	"unicode"
+
+	"github.com/mitchellh/go-wordwrap"
+	"golang.org/x/crypto/ssh/terminal"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/cli/clibase"
+	"github.com/coder/coder/cli/cliui"
+)
+
+//go:embed help.tpl
+var helpTemplateRaw string
+
+type optionGroup struct {
+	Name        string
+	Description string
+	Options     clibase.OptionSet
+}
+
+func ttyGetSize() int {
+	width, _, err := terminal.GetSize(0)
+	// Even in tests, we want to return the default width because
+	// the PTY width changes across operating systems for some
+	// unknown reason.
+	if err != nil || flag.Lookup("test.v") != nil {
+		return 80
+	}
+	return width
+}
+
+// wrapTTY wraps a string to the width of the terminal, or 80 no terminal
+// is detected.
+func wrapTTY(s string) string {
+	return wordwrap.WrapString(s, uint(ttyGetSize()))
+}
+
+var usageTemplate = template.Must(
+	template.New("usage").Funcs(
+		template.FuncMap{
+			"wrapTTY": func(s string) string {
+				return wrapTTY(s)
+			},
+			"trimNewline": func(s string) string {
+				return strings.TrimSuffix(s, "\n")
+			},
+			"indent": func(body string, tabs int) string {
```

Comment:
Hmm, why `tabs` over `count`? `tabs` is more clear to me that we're talking about multiple spaces.

Reply:
To me, `tabs` would suggest we’re using the actual tab character `\t`. Calling it `spaces` would be clearer but would require dropping the embedded size so that you do `indent(4)` instead of `indent(1)`, which would be a good chance too, IMO.